AdminWebsite/AdminWebsite/ClientApp/src/app/popups/confirm-booking-failed-popup/confirm-booking-failed-popup.component.spec.ts

```diff
@@ -23,17 +24,26 @@ describe('SaveFailedPopupComponent', () => {
         component = fixture.componentInstance;
         de = fixture.debugElement;
         buttonTryAgain = de.query(By.css('#btnTryAgain'));
-        buttonCancel = de.query(By.css('#btnCancel'));
+        headingText = de.query(By.css('.govuk-heading-m'));
+        bodyText = de.query(By.css('#content-text'));
         fixture.detectChanges();
     });
 
     it('should create', () => {
         expect(component).toBeTruthy();
     });
 
-    it('should emit event when the ok button is clicked', () => {
+    it('should emit event when the close button is clicked', () => {
         spyOn(component.close, 'emit');
         buttonTryAgain.nativeElement.click();
         expect(component.close.emit).toHaveBeenCalled();
     });
+
+    it('should have heading Booking confirmation failed!', () => {
+        expect(headingText.nativeElement.innerText).toEqual('Booking confirmation failed!');
```

Comment:
I don't think we should be testing text content in unit test, can we instead check if the element with an id/data-handler exist instead?

Reply:
it is tested because the previous implementation had different text and bug was raised. we made sure the text is correct.